lib/python/picongpu/pypicongpu/laser.py

```diff
@@ -102,3 +108,172 @@ def _get_serialized(self) -> dict:
+
+        LINEAR = 1
+        CIRCULAR = 2
+
+        def get_cpp_str(self) -> str:
+            """retrieve name as used in c++ param files"""
+            cpp_by_ptype = {
+                PlaneWaveLaser.PolarizationType.LINEAR: "Linear",
+                PlaneWaveLaser.PolarizationType.CIRCULAR: "Circular",
+            }
+            return cpp_by_ptype[self]
+
+    wavelength = util.build_typesafe_property(float)
+    """wave length in m"""
+    duration = util.build_typesafe_property(float)
+    """duration in s (1 sigma)"""
+    focus_pos = util.build_typesafe_property(typing.List[float])
+    """focus position vector in m"""
+    phase = util.build_typesafe_property(float)
+    """phase in rad, periodic in 2*pi"""
+    E0 = util.build_typesafe_property(float)
+    """E0 in V/m"""
+    pulse_init = util.build_typesafe_property(float)
+    """laser will be initialized pulse_init times of duration (unitless)"""
+    propagation_direction = util.build_typesafe_property(typing.List[float])
+    """propagation direction(normalized vector)"""
+    polarization_type = util.build_typesafe_property(PolarizationType)
+    """laser polarization"""
+    polarization_direction = util.build_typesafe_property(typing.List[float])
+    """direction of polarization(normalized vector)"""
+    laser_nofocus_constant_si = util.build_typesafe_property(float)
+    """constant for plane wave laser without focus (unitless)"""
+    huygens_surface_positions = util.build_typesafe_property(typing.List[typing.List[int]])
+    """Position in cells of the Huygens surface relative to start/
+       edge(negative numbers) of the total domain"""
+
+    def _get_serialized(self) -> dict:
+        return {
+            "wave_length_si": self.wavelength,
+            "pulse_duration_si": self.duration,
+            "focus_pos_si": list(map(lambda x: {"component": x}, self.focus_pos)),
+            "phase": self.phase,
+            "E0_si": self.E0,
+            "pulse_init": self.pulse_init,
+            "propagation_direction": list(map(lambda x: {"component": x}, self.propagation_direction)),
+            "polarization_type": self.polarization_type.get_cpp_str(),
+            "polarization_direction": list(map(lambda x: {"component": x}, self.polarization_direction)),
+            "laser_nofocus_constant_si": self.laser_nofocus_constant_si,
+            "huygens_surface_positions": {
+                "row_x": {
+                    "negative": self.huygens_surface_positions[0][0],
+                    "positive": self.huygens_surface_positions[0][1],
+                },
+                "row_y": {
+                    "negative": self.huygens_surface_positions[1][0],
+                    "positive": self.huygens_surface_positions[1][1],
+                },
+                "row_z": {
+                    "negative": self.huygens_surface_positions[2][0],
+                    "positive": self.huygens_surface_positions[2][1],
+                },
+            },
+        }
```

Comment:
I guess this applies to everywhere we define the surfaces. What do we do if we have more than one laser? Huygens surface position are the same for all lasers in PIConGPU.

Reply:
currently our PICMI implementation is still limited to one or no laser, so a future problem @chillenzer